tests/rados/test_rados_watcher_cleanup.py

```diff
@@ -0,0 +1,1013 @@
+                    all_created_images.append(
+                        image_name
+                    )  # Add to the list for final cleanup
+                    try:
+                        _add_watchers(
+                            client_obj=client,
+                            pool=pool,
+                            rados_orchestrator_obj=rados_obj,
+                            image_name=image_name,
+                            mapped_devices=mapped_devices,
+                        )
+                    except Exception as e:
+                        log.error(e)
+                        log.error(
+                            f"Failed to create image and watcher for {image_name}"
+                        )
+                        return 1  # Fail test if watcher creation fails
+
+        log.info(
+            "\n--- Phase 1: All images created, mapped, mounted, and watchers verified. Proceeding to cleanup. ---"
+        )
+
+        # --- Phase 2: Delete Watchers based on various methods (excluding blacklist) ---
+        log.info(
+            "\n--- Phase 2: Initiating Watcher Cleanup Verification (Excluding Blacklist) ---"
+        )
+        cleanup_methods: List[str] = [
+            "unmap",
+            "umount",
+            "restart",
+            "crash",
+        ]
+        # Ensure we have enough images for each method per client
+        if num_images_to_create_per_client < len(cleanup_methods):
+            raise Exception(
+                f"Not enough images created ({num_images_to_create_per_client}) for all"
+                "cleanup methods ({len(cleanup_methods)}). Increase num_images_to_create_per_client."
+            )
+
+        # Iterate through mapped devices to apply cleanup methods
+        for client_hostname, images_info in list(mapped_devices.items()):
+            client = get_host_obj_from_hostname(
+                hostname=client_hostname, rados_obj=rados_obj
+            )
+
+            if not client:
+                log.warning(
+                    "Client %s not found, skipping cleanup for its images.",
+                    client_hostname,
+                )
+                continue
+
+            # Associate images with methods directly for this phase
+            # Assuming images are named img_*_0, img_*_1, etc.
+            images_for_method_test = sorted(
+                images_info.keys()
+            )  # Ensure consistent order
+            if len(images_for_method_test) < len(cleanup_methods):
+                log.warning(
+                    f"Not enough images ({len(images_for_method_test)}) for all non-blacklist"
+                    "cleanup methods ({len(cleanup_methods)}) on client {client_hostname}. "
+                    "Some methods might be skipped."
+                )
+                # Adjusting cleanup_methods to match available images
+                cleanup_methods_to_run = cleanup_methods[: len(images_for_method_test)]
```

Comment:
This one check is enough for all possible variations, please remove if else conditional checks